include/range/v3/view/any_view.hpp

```diff
@@ -102,155 +176,106 @@ namespace ranges
                 {
-                    --it_.get();
+                    --it_;
                 }
                 void advance(std::ptrdiff_t n) // override-ish
                 {
-                    it_.get() += n;
+                    it_ += n;
                 }
                 std::ptrdiff_t distance_to(
-                    any_cursor_interface<Ref, category::random_access> const &that) const // override-ish
+                    any_cursor_interface<Ref, Cat> const &that_) const // override-ish
                 {
-                    any_cursor_impl const *pthat =
-                        polymorphic_downcast<any_cursor_impl const *>(&that);
-                    return static_cast<std::ptrdiff_t>(pthat->it_.get() - it_.get());
+                    auto &that = polymorphic_downcast<any_cursor_impl const &>(that_);
+                    return static_cast<std::ptrdiff_t>(that.it_ - it_);
                 }
             };
 
             struct any_sentinel_interface
             {
-                virtual ~any_sentinel_interface() = default;
-                virtual bool equal(any_object const &) const = 0;
-                virtual any_sentinel_interface *clone() const = 0;
+                virtual bool at_end(any_ref) const = 0;
+            protected:
+                ~any_sentinel_interface() = default;
             };
 
-            template<typename S, typename I>
-            struct any_sentinel_impl
-              : any_sentinel_interface
-            {
-            private:
-                S s_;
-            public:
-                any_sentinel_impl() = default;
-                any_sentinel_impl(S s)
-                  : s_(std::move(s))
-                {}
-                bool equal(any_object const &that) const override
-                {
-                    object<I> const *pthat = polymorphic_downcast<object<I> const *>(&that);
-                    return s_ == pthat->get();
-                }
-                any_sentinel_impl *clone() const override
-                {
-                    return new any_sentinel_impl{s_};
-                }
-            };
-
-            template<typename Ref, category Cat>
-            struct any_cursor;
-
             struct any_sentinel
             {
-            private:
-                template<typename, category>
-                friend struct any_cursor;
-                std::unique_ptr<any_sentinel_interface> ptr_;
-            public:
                 any_sentinel() = default;
-                template<typename Rng,
-                    CONCEPT_REQUIRES_(!Same<detail::decay_t<Rng>, any_sentinel>()),
-                    CONCEPT_REQUIRES_(InputRange<Rng>())>
-                any_sentinel(Rng &&rng, end_tag)
-                  : ptr_{new any_sentinel_impl<sentinel_t<Rng>, iterator_t<Rng>>{
-                        end(rng)}}
+                constexpr explicit any_sentinel(any_sentinel_interface const &view) noexcept
+                  : view_{&view}
```

Comment:
> `any_cursor`s don't know where the `view` is, but `any_sentinel`s do. 

Yes, this is ironically opposite the case for the input view implementation: `any_input_view_cursor` (which I should really rename to `any_input_cursor`) holds a pointer to the view which `default_sentinel` obviously does not.